sklearn/inspection/_partial_dependence.py

```diff
@@ -592,13 +593,20 @@ def plot_partial_dependence(estimator, X, features, feature_names=None,
         # regression and binary classification
         target_idx = 0
 
-    X = check_array(X)
+    # Use check_array only on lists and other non-array-likes / sparse. Do not
+    # convert DataFrame into a NumPy array.
+    if not(hasattr(X, '__array__') or sparse.issparse(X)):
+        X = check_array(X, force_all_finite='allow-nan', dtype=np.object)
```

Comment:
I feel like at some point this should be inside the `check_array`. Also, why not pass `accept_sparse` to check_array and not check it here?

Reply:
> I feel like at some point this should be inside the check_array

Agreed. I think that @jorisvandenbossche intended a PR on this a while ago.

> Also, why not pass accept_sparse to check_array and not check it here?

I think that the idea was to delegate the check to the underlying pipeline.
In fact, I am not sure that we need to make any checking at all. If we only need to get a column, `_safe_indexing` should be smart enough to deal with list.

Reply:
Actually the next line will fail `n_features=X.shape[0]` so we need something else than a list.